### PR TITLE
Remove dead code associated with old roles

### PR DESCRIPTION
**Pull request by benhalpern**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Old code related to roles that don't do anything.

### CODE REVIEW

File: app/models/role.rb [hunk 1]
```diff
@@ -16,7 +15,6 @@ class Role < ApplicationRecord
     tech_admin
     trusted
     warned
-    workshop_pass
```
Comment (lightalloy) [inline app/models/role.rb:19]: Also locales, `role_spec`, and `seeds.rb`

File: app/policies/authorizer.rb [hunk 2]
```diff
@@ -165,10 +165,6 @@ def warned?
       has_role?(:warned)
     end
 
-    def workshop_eligible?
```
Comment (lightalloy) [inline app/policies/authorizer.rb:168]: There is a test in `models/user_spec.rb` which expects this method, let's delete the test.

**Review by lightalloy — CHANGES_REQUESTED** [review]
It feels good to remove unused stuff :scissors: 
Let's remove all references of `workshop_pass` and `mod_relations_admin`, I have described in the comments.
You can find them just by searching for `workshop_pass`, `mod_relations_admin`, `workshop_eligible` in the codebase.

**Comment by benhalpern** [comment]
Thanks for the input @lightalloy

**Review by lightalloy — APPROVED** [review]
Thanks for implementing changes, I have removed the related locales as well.

**Pull request merged**